src/data/course-discovery/api.ts

```diff
@@ -13,12 +13,16 @@ import { CourseDiscoveryResponse } from './types';
 export const fetchCourseDiscovery = async (
   pageSize = DEFAULT_PAGE_SIZE,
   pageIndex = DEFAULT_PAGE_INDEX,
+  enableCourseSortingByStartDate = false,
 ): Promise<CourseDiscoveryResponse> => {
+  const formData = new FormData();
+
+  formData.append('page_size', String(pageSize));
+  formData.append('page_index', String(pageIndex));
+  formData.append('enable_course_sorting_by_start_date', String(enableCourseSortingByStartDate));
+
   const { data } = await getAuthenticatedHttpClient()
-    .post(getCourseDiscoveryUrl(), {
-      page_size: pageSize,
-      page_index: pageIndex,
-    });
+    .post(getCourseDiscoveryUrl(), formData);
```

Comment:
[inform/question]: During testing, I realized that the `course_discovery` API only returns courses for which `enrollment_end` is either not specified or the enrollment date for the course has not yet expired. We will address this issue on the backend side.

But, it seems strange to me that the legacy page displayed courses for which the enrollment date had passed and a student (or new student), upon seeing this course, would not be able to register for it.

Do you have any thoughts on this?